File: golang/src/domain/produto_test.go
package domain

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestPeso_CalcularDensidade(t *testing.T) {
	produto := Produto{
		Codigo:    "2323-AFSRDHDKDL-456",
		Descricao: "Camera",
		Dimensao:  Dimensao{20, 15, 10},
		Peso:      Peso{1},
	}

	assert.Equal(t, 0.00999, produto.getIndiceCalculoFrete())
}

File: node/test/domain/Produto.test.ts
import Dimensao from "../../src/domain/Dimensao"
import Peso from "../../src/domain/Peso"
import Produto from "../../src/domain/Produto"

test("Deve calcular o indice de calculo para o frete com base no volume e densidade", () => {

    const produto = new Produto("2323-AFSRDHDKDL-456", "Camera", new Dimensao(20,15,10), new Peso(1)) 

    expect(0.00999).toEqual(produto.getIndiceCalculoFrete())
})